### DISCUSSION TITLE
Adding New Environment Variable in Sunbird Ed Backend

### DISCUSSION THREAD

**Discussion by ManojNathIC** [Sunbird-Ed/Community#696] [Bugs] [answered]
Hi Team,

I'm encountering an issue with adding a new environment variable that works in my local development environment but not in the dev environment.

Is there a specific procedure or configuration required to add environment variables in the Sunbird Ed backend?

I appreciate any insights. Thank you!

@vinukumar-vs , @rajeevsathish, @AmiableAnil , @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, @coolbung, @komalm @paritshivani @manishakapadnis

**Comment by rajeevsathish** [comment]
@gandham-santhosh can you please help them how to add a new variable to the devops yml file

**Reply by gandham-santhosh** [reply, marked as answer]
@ManojNathIC which version of SunbirdEd you are using?. Follow below steps 

- Fork https://github.com/project-sunbird/sunbird-devops repository to your github org where you are doing development
- Select the branch based on the version you are using. If you are using Ed6.0 then select release-6.0.0 branch
- Update the below env file and add your new environment variable
https://{your_forked_github_repo_url}/sunbird-devops/blob/release-6.0.0/ansible/roles/stack-sunbird/templates/sunbird_player.env
- Initialise the values in your private repo common.yaml 
- Update Portal Deploy Jenkins job to point to your forked sunbird-devops repo
- Deploy the portal service to your dev env

BTW, is this some new feature you are developing which requires to add new variable to portal? Just want to understand the objective of adding new variable

**Reply by ManojNathIC** [reply]
@gandham-santhosh Yes, we've new requirements for developing a new service that requires an external database(we are not using the Sunbird database). To ensure smooth integration and functionality for the database, we need to add a new environment variable for the database credentials in the env file.

**Reply by gandham-santhosh** [reply]
@ManojNathIC Sure. Let us know if you are able to follow the steps and add a variable.

**Comment by anisha147** [comment]
Hi @ManojNathIC 

We hope you successfully added a new environment variable for the database credentials in the env file. Do you have any more queries on this topic? If not, could you please close this thread from your end?

cc: @gandham-santhosh

**Comment by jayesh9747** [comment]
@gandham-santhosh   hello sir , i want to set-up ml-survey service in local machine  but i faced this issue  , how to resolve this ? 
ml-survey-service repo link : https://github.com/shikshalokam/ml-survey-service?tab=readme-ov-file 
![image](https://github.com/Sunbird-Ed/Community/assets/123955234/efd2c496-14e1-4912-96d5-e458163e38e1)

**Comment by HarishGangula** [comment]
@jayesh9747 ml survey is not part of the Ed please raise issue in the respective blocks discussion thread.
